src/System.Private.CoreLib/src/System/Runtime/CompilerServices/ConditionalWeakTable.cs

```diff
@@ -270,12 +270,12 @@ private void CreateEntry(TKey key, TValue value)
 
         #region Private Data Members
         //--------------------------------------------------------------------------------------------
-        // Entry can be in one of three states:
+        // Entry can be in one of four states:
         //
-        //    - Linked into the freeList (_freeList points to first entry)
+        //    - Unused (stored at an index below _firstEntry)
```

Comment:
I think it would be a lot easier to follow what is going on if the table is filled in from entry 0. I do not think you are gaining anything by filling it from the top (two instructions saved in HasCapacity do not matter).